src/praisonai-agents/praisonaiagents/tools/decorator.py

```
@@ -214,6 +214,13 @@ def decorator(fn: Callable) -> FunctionTool:
             availability=availability
         )
         
+        # Validate the tool at creation time for early error detection
+        try:
+            tool_instance.validate()
+            tool_instance.validate_schema_roundtrip()
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)